### PR TITLE
[RTM] Add news sorting ability

### PR DESCRIPTION
**Pull request by fritzmg**
This implements #12 

I added it for the newslist as well as the newsarchive.

### CODE REVIEW

**Comment by fritzmg** [comment]
I have changed the field name to `news_sorting`, which is shorter and more general (since I also implemented the functionality into the news archive module).

File: src/Resources/contao/modules/ModuleNewsArchive.php [hunk 1]
```diff
@@ -186,14 +186,39 @@ protected function compile()
...
+
+			case 'sort_headline_asc':
+				$arrOptions['order'] = "$t.headline ASC";
+				break;
+
+			case 'sort_headline_desc':
+				$arrOptions['order'] = "$t.headline DESC";
+				break;
+
+			case 'sort_random':
```
Comment (leofeyer) [inline src/Resources/contao/modules/ModuleNewsArchive.php:207]: Do you really think that randomly sorted news is something anyone needs?

Comment (fritzmg) [reply src/Resources/contao/modules/ModuleNewsArchive.php:207]: Yes, we needed it a couple of times at least. Currently in three projects as well.

**Comment by dmolineus** [comment]
Also had projects with the requirement of random sorted news/events. Often is required when a set of highlights are teasered on the homepage.

**Comment by fritzmg** [comment]
Any further changes required? Otherwise I'd consider this RTM.

**Comment by leofeyer** [comment]
You should mark it `RTM` when **you** think it is ready to merge. 😄

**Comment by fritzmg** [comment]
Right 😁

**Comment by leofeyer** [comment]
I just realized that there is a `tl_module.news_order` field already, which is currently used in the news archive menu. We should use the field in the news list, too, instead of adding yet another field.

**Comment by leofeyer** [comment]
I'll adjust the PR accordingly.

**Comment by fritzmg** [comment]
Indeed. I forgot about that as well.

**Comment by leofeyer** [comment]
Merged in 0241e9aa511eb8ce8d6a53d55663497b83c78c54.

**Pull request closed without merging**